src/createrepo_c.c

```diff
@@ -1190,6 +1190,21 @@ main(int argc, char **argv)
         cr_repomd_record_rename_file(prestodelta_rec, NULL);
     }
 
+    if (cmd_options->set_timestamp_to_revision) {
+        // validated already in cmd_parser.c:check_arguments
+        gint64 revision = strtoll(cmd_options->revision, NULL, 0);
```

Comment:
What about to support --revision and --set-timestamp-to-revision independently?
I think it's a valid use case to specify --set-timestamp-to-revision only when I want to set the revision and timestamps to the current time.

(pseudo-code follows)
```
// use user-defined revision or current time
revision = options->revision ? options->revision : time(NULL))

if options->set_timestamp_to_revision {
   // allways set the timestamps if the option is present (regardless if --revision is specified)
   cr_repomd_record_set_timestamp(...)
   ...
}

```
this may require a small change in cr_xml_dump_repomd_body()
there's a fallback to time(NULL) and it may not be needed any more as repomd->revision will allways be set

Reply:
IMO if you need to control details like revision number, or timestamps, explicit is better than implicit. It's easy enough to specify `--revision $(date +%s)`.